### ISSUE TITLE
The Best top-1 acc about cifar10-250 is 89.60.

### ISSUE THREAD

**Issue by xiaopingzeng** [kekmodel/FixMatch-pytorch#1]
Thanks for your reimplement, but I get the Best top-1 acc about cifar10-250 is 89.60(the result of the paper is 94.93 ± 0.65).

**Comment by kekmodel** [comment]
Right. I can't find where the problem is. I need your feedback.

**Comment by NovaMind-Z** [comment]
Hi, have you found the problem that causes the difference between the results in the paper and your code?

**Comment by kekmodel** [comment]
https://github.com/kekmodel/FixMatch-pytorch/commit/e867808ab317f64a449988ecb60c8271b677bcb3#diff-e44f4a60e89e820dde9bb27afa634965

**Comment by leekyungmoon** [comment]
Thanks for great work,

What do you think of the reason for the 1% gap between the figures from the paper and yours with the updated code above?

It may be caused by the random split between labeled data and unlabeled data, right?

**Comment by JasonEricZhan** [comment]
> Thanks for great work,
> 
> What do you think of the reason for the 1% gap between the figures from the paper and yours with the updated code above?
> 
> It may be caused by the random split between labeled data and unlabeled data, right?

I guess the sampling method can affect. To be more specific, from my experiments, some implement with sampling at each batch can get better result, and in this seems just expand data and sampling at one time.

Yet, I am not very sure about the real implement of the paper in this part.

**Comment by JasonEricZhan** [comment]
I guess the sampling method can affect. To be more specific, from my experiments, some implement with sampling at each batch can get better result. However, the implement here expands data and sampling at one time.

Yet, I am not very sure about the real implement of the paper in this part.

**Comment by JasonEricZhan** [comment]
> Thanks for great work,
> 
> What do you think of the reason for the 1% gap between the figures from the paper and yours with the updated code above?
> 
> It may be caused by the random split between labeled data and unlabeled data, right?

I guess the sampling method can affect. To be more specific, from my experiments, some implement with sampling at each batch can get better result. However, the implement here expands data and sampling at one time.

Yet, I am not very sure about the real implement of the paper in this part.

**Comment by wangtingwei1993** [comment]
@JasonEricZhan. I am not quite sure why they need to expand the number labeled data (40, 250, 4000 for cifar10) to 65536(default setting). Could you please explain this form me?
Thanks a lot !

**Comment by Chen-Song** [comment]
@JasonEricZhan Could you make details about what you said 'from my experiments, some implement with sampling at each batch can get better result'. Thanks.

**Comment by bkj** [comment]
Was anyone able to get an accuracy close to the ~94% reported in the paper?  I'm running
```
python train.py \
     --dataset        cifar10       \
     --num-labeled    250           \
     --arch           wideresnet    \
     --batch-size     64            \
     --lr             0.03          \
     --out            cifar10@250-1GPU
```
and the test accuracy curve looks like
<img width="363" alt="Screen Shot 2020-09-18 at 9 41 55 AM" src="https://user-images.githubusercontent.com/6086781/93604374-4ae12800-f993-11ea-8719-b544eb7d8c15.png">

Have only trained for ~ 10% of the full schedule, but it doesn't look like it's going to go up another 6%.  Anyone have any thoughts?

**Comment by kekmodel** [comment]
There was a problem with the EMA. Now it reproduces well! Thank you.
